homeassistant/components/automation/__init__.py

```diff
@@ -394,13 +397,16 @@ def referenced_entities(self):
         _LOGGER.info("Executing %s", self._name)
 
         try:
-            await self.action_script.async_run(variables, trigger_context)
-        except Exception as err:  # pylint: disable=broad-except
-            self.action_script.async_log_exception(
-                _LOGGER, f"Error while executing automation {self.entity_id}", err
+            await self.action_script.async_run(
+                variables,
+                trigger_context,
+                _LOGGER,
+                f"Error while executing automation {self.entity_id}",
```

Comment:
This is too specific. Instead we should pass `f"automation {self.entity_id}"` to the constructor so that the script can use it in all log messages. 

Actually, we should just pass a unique logger instead of a name `logging.getLogger(f"{__name__}.{self.entity_id}")`. Also the logger should be passed to the constructor, not each rnu.

Reply:
Good ideas. Thanks! Will do.